server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java

```diff
@@ -40,22 +49,62 @@
    public RestRequestHandler(RestServer restServer) {
       this.restServer = restServer;
       this.configuration = restServer.getConfiguration();
-   }
-
-   @Override
-   public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
-      authenticationHandler = ctx.pipeline().get(AuthenticationHandler.class);
-      super.channelRegistered(ctx);
+      this.authenticator = configuration.authentication().enabled() ? configuration.authentication().authenticator() : null;
    }
 
    @Override
    public void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
       restAccessLoggingHandler.preLog(request);
       NettyRestRequest restRequest = new NettyRestRequest(request);
-      if (authenticationHandler != null) {
-         restRequest.setSubject(authenticationHandler.getSubject());
+
+      LookupResult invocationLookup = restServer.getRestDispatcher().lookupInvocation(restRequest);
+
+      if (authenticator == null || isAnon(invocationLookup)) {
```

Comment:
as an enhancement we might also introduce a sourceIP whitelist

Reply:
I was looking at it actually. @ryanemerson would it be ok if Health calls are restricted to localhost?

Reply:
@wburns true, although load balancers usually need only a 200 OK response without content, I will add it to the server resource where it currently returns only the version

Reply:
> I was looking at it actually. @ryanemerson would it be ok if Health calls are restricted to localhost?

This should be fine, as the probes are always executed in the image themselves. However, I do like Tristan's idea of a white list as I can imagine external users not wanting to expose anything to anonymous users.